src/dodal/devices/i22/nxsas.py

```diff
@@ -0,0 +1,180 @@
+from typing import Dict, Tuple
+
+from bluesky.protocols import Reading
+from event_model.documents.event_descriptor import DataKey
+from ophyd_async.core import (
+    ConfigSignal,
+    DirectoryProvider,
+    StandardReadable,
+    soft_signal_r_and_setter,
+)
+from ophyd_async.epics.areadetector import AravisDetector, PilatusDetector
+from ophyd_async.epics.areadetector.aravis import AravisController
+
+
+class NXSasMetadataHolder(StandardReadable):
+    ValueAndUnits = Tuple[float, str]
+    """
+    Required fields for NXDetectors that are used in an NXsas application definition
+    """
+
+    def __init__(
+        self,
+        distance: ValueAndUnits,
+        x_pixel_size: ValueAndUnits,
+        y_pixel_size: ValueAndUnits,
+        beam_center_x: ValueAndUnits | None = None,
+        beam_center_y: ValueAndUnits | None = None,
+        aequetorial_angle: ValueAndUnits | None = None,
+        azimuthal_angle: ValueAndUnits | None = None,
+        polar_angle: ValueAndUnits | None = None,
+        rotation_angle: ValueAndUnits | None = None,
+        threshold_energy: ValueAndUnits | None = None,
+        serial_number: str | None = None,
+        sensor_material: str | None = None,
+        sensor_thickness: ValueAndUnits | None = None,
+        description: str | None = None,
+        type: str | None = None,
+        prefix: str = "",
+        name: str = "",
+    ):
+        with self.add_children_as_readables(ConfigSignal):
+            self.distance, _ = soft_signal_r_and_setter(
+                float, distance[0], units=distance[1]
+            )
+            self.x_pixel_size, _ = soft_signal_r_and_setter(
+                float, x_pixel_size[0], units=x_pixel_size[1]
+            )
+            self.y_pixel_size, _ = soft_signal_r_and_setter(
+                float, y_pixel_size[0], units=y_pixel_size[1]
+            )
+            if polar_angle is not None:
+                self.polar_angle, _ = soft_signal_r_and_setter(
+                    float, polar_angle[0], units=polar_angle[1]
+                )
+            else:
+                self.polar_angle = None
+            if azimuthal_angle is not None:
+                self.azimuthal_angle, _ = soft_signal_r_and_setter(
+                    float, azimuthal_angle[0], units=azimuthal_angle[1]
+                )
+            else:
+                self.azimuthal_angle = None
+            if rotation_angle is not None:
+                self.rotation_angle, _ = soft_signal_r_and_setter(
+                    float, rotation_angle[0], units=rotation_angle[1]
+                )
+            else:
+                self.rotation_angle = None
```

Comment:
make this a oneliner: `self.rotation_angle, _ = soft_signal_r_and_setter(float, rotation_angle[0], units=rotation_angle[1]) if rotation_angle is not None else (None, None)
`

Reply:
and others too

Reply:
This is an established pattern for soft signals for the moment.
I agree that it's verbose, but unfortunately it seems to be required for type checking
https://github.com/DiamondLightSource/dodal/blob/0e15a5ffd5bf5b8b9ed09a81c6d37608af0272c9/src/dodal/devices/i22/fswitch.py#L54

Reply:
ok I see, I didn't realize that.